File: VTI_Galaxy_BE/src/main/java/demo/services/interfaceClass/GalaxyService.java
package demo.services.interfaceClass;

import demo.modal.dto.GalaxyDto;
import demo.modal.request.GalaxyRequest;

import java.util.List;

public interface GalaxyService {
    // get
    List<GalaxyDto> getGalaxies();
    GalaxyDto getGalaxyById(int id);

    // post
    GalaxyDto addGalaxy(GalaxyRequest request) throws Exception;

    // put
    GalaxyDto updateGalaxy(GalaxyRequest request, int id);

    // delete(closed)
    GalaxyDto closedGalaxy(int id);
}

File: VTI_Glaxy_FE/src/services/galaxyService.js
import axiosClient from "./axiosClient";

const galaxyService = {
   fetchGalaxy(){
      return axiosClient.get("/getGalaxies")
   },
   fetchGalaxyById(galaxyId){
      return axiosClient.get(`/getGalaxyById?galaxyId=${galaxyId}`)
   },
   createGalaxy(galaxyRequest){
      return axiosClient.post("/postGalaxy", galaxyRequest)
   },
   updateGalaxy(galaxyId, galaxyRequest){
      return axiosClient.put(`/putGalaxy?galaxyId=${galaxyId}`, galaxyRequest)
   },
   deleteGalaxy(galaxyId){
      return axiosClient.put(`/putStatusGalaxy?galaxyId=${galaxyId}`)
   }
}

export default galaxyService;